### PR TITLE
Refactor navigate back in Assessment Editors

### PR DESCRIPTION
**Pull request by jpbernius**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
We use the same logic to navigate back to the tutor exercise dashboard for text, modeling, and file upload exercises.
Here we refactor this into one common method.

### Steps for Testing

1. Assess text, modeling, and file upload exercises
2. In the editor, try to navigate back to the tutor exercise dashboard using the ⬅️ button.
3. Repeat for an exam.

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/navigate-back.util.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Location } from '@angular/common';
+import { Router } from '@angular/router';
+import { StudentParticipation } from 'app/entities/participation/student-participation.model';
+import { Exercise } from 'app/entities/exercise.model';
+import { Submission } from 'app/entities/submission.model';
+
+export function asseessmentNavigateBack(location: Location, router: Router, exercise: Exercise | null, submission: Submission | null) {
```
Comment (JonasPetry) [inline src/main/webapp/app/exercises/shared/navigate-back.util.ts:7]:
```suggestion
export function assessmentNavigateBack(location: Location, router: Router, exercise: Exercise | null, submission: Submission | null) {
```

Comment (jpbernius) [reply src/main/webapp/app/exercises/shared/navigate-back.util.ts]: :+1:

File: src/main/webapp/app/exercises/shared/navigate-back.util.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Location } from '@angular/common';
+import { Router } from '@angular/router';
+import { StudentParticipation } from 'app/entities/participation/student-participation.model';
+import { Exercise } from 'app/entities/exercise.model';
+import { Submission } from 'app/entities/submission.model';
+
+export function assessmentNavigateBack(location: Location, router: Router, exercise: Exercise | null, submission: Submission | null) {
```
Comment (krusche) [inline src/main/webapp/app/exercises/shared/navigate-back.util.ts:7]: please add some documentation for the different cases

**Comment by HanyaElhashemy** [comment]
One thing I just noticed: there is a back button when you create a sample submission and its sample assessment. Currently, the back button throws a console error for modeling exercises when you (as an instructor) create a sample submission then create a sample assessment then try to navigate back through the button to the "create/edit exercise page". Could we maybe reuse this method there?
(the error is caused by the back() method because the examMode boolean is set wrong I guess but I suggest here using the assessmentNavigateBack() method instead of back())
image

**Review by fde312 — APPROVED** [review]
Works as expected and code looks also good to me! 👍 

I found the same issue as @HanyaElhashemy, but I think this can also be done in a follow up.
I created a issue for this: https://github.com/ls1intum/Artemis/issues/2039

**Review by krusche — APPROVED** [review]

**Pull request merged**